### PR TITLE
Add the `CallWebView` logs to our logging stack

### PR DESCRIPTION
**Pull request by jmartinesp**
## Content

Intercept the logs for the Element Call webview and log them using Timber so they go through our logging system.

## Motivation and context

Element Call logs were lost in nightly and release versions of the app since they're not printed to the logcat or saved to disk, so it's very difficult to debug issues unless it's in a debug version.

## Tests

- Open a call using EXA.
- Check the logcat, send a rageshake or check the logs using the bug report file viewer.
- Check there are `[WebView]` prefixed logs inside.

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): 14

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR

### CODE REVIEW

File: features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/CallScreenView.kt [hunk 1]
```diff
@@ -189,6 +191,11 @@ private fun WebView.setup(
         override fun onPermissionRequest(request: PermissionRequest) {
             onPermissionsRequested(request)
         }
+
+        override fun onConsoleMessage(consoleMessage: ConsoleMessage?): Boolean {
+            consoleMessage?.let { Timber.d("[WebView]: ${it.message()}") }
```
Comment (bmarty) [inline features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/CallScreenView.kt:196]: Could maybe have a better mapping using other provided field:

```kotlin
override fun onConsoleMessage(consoleMessage: ConsoleMessage): Boolean {
    val priority = when (consoleMessage.messageLevel()) {
        ConsoleMessage.MessageLevel.ERROR -> Log.ERROR
        ConsoleMessage.MessageLevel.WARNING -> Log.WARN
        else -> Log.DEBUG
    }
    Timber.tag("[WebView]").log(
        priority = priority,
        message = consoleMessage.sourceId() + ":" + consoleMessage.lineNumber() + " " + consoleMessage.message(),
    )
    return true
}
```

**Review by bmarty — APPROVED** [review]
Thanks, but are we sure that the log from the WebView will not contain any sensitive data?

**Comment by jmartinesp** [comment]
> Thanks, but are we sure that the log from the WebView will not contain any sensitive data?

According to @toger5 they don't contain sensitive data, the only thing that could be considered 'sensitive' are user ids, but that should be fine to log.

File: features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/CallScreenView.kt (hunk 1 above)
Comment (jmartinesp) [reply features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/CallScreenView.kt]: https://github.com/element-hq/element-x-android/pull/3637/commits/2f2d8515059a558bddcad46b0ddb75ccceea5b05

**Pull request merged**